pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -373,6 +405,89 @@ func (r *SystemReconciler) systemAppDeploymentResourceMutator(desired, existing
 	return update, nil
 }
 
+func (r *SystemReconciler) systemZyncEnvVarMutator(desired, existing *k8sappsv1.Deployment) (bool, error) {
+	update := false
+
+	// Reconcile ZYNC_AUTHENTICATION_TOKEN env var
+	for idx := range existing.Spec.Template.Spec.Containers {
+		tmpChanged := helper.EnvVarReconciler(
+			desired.Spec.Template.Spec.Containers[idx].Env,
+			&existing.Spec.Template.Spec.Containers[idx].Env,
+			"ZYNC_AUTHENTICATION_TOKEN")
+		update = update || tmpChanged
+	}
+
+	return update, nil
+}
+
+func (r *SystemReconciler) routesExist() (bool, error) {
```

Comment:
this seems like something apim status reconciler could use as well no?

Reply:
Good call, I'll move the function

Reply:
Fixed in latest commit